### ISSUE TITLE
[Question] Trained Models not give best rewards (Hugging Face Models) 

### ISSUE THREAD

**Issue by indramal** [DLR-RM/stable-baselines3#1128]
### ❓ Question

I am trying to load and test trained models. I got trained model from Hugging face and I evaluate performance and that give `mean_reward=6.60 +/- 4.758150901348128`. Do I need to set Hyperparameters? If so, how can I do it?  

Code:

```
import gym

from huggingface_sb3 import load_from_hub
from stable_baselines3 import PPO
from stable_baselines3.common.evaluation import evaluate_policy

from google.colab.patches import cv2_imshow

checkpoint = load_from_hub(
    repo_id="sb3/ppo-BreakoutNoFrameskip-v4",
    filename="ppo-BreakoutNoFrameskip-v4.zip",
)
model = PPO.load(checkpoint)

eval_env = make_atari_env("Breakout-v4", n_envs=4, seed=0)
eval_env = VecFrameStack(eval_env , n_stack=4)

mean_reward, std_reward = evaluate_policy(
    model, eval_env, n_eval_episodes=10, deterministic=True, warn=False
)
print(f"mean_reward={mean_reward:.2f} +/- {std_reward}")
```

Hyperparameters:

> OrderedDict([('batch_size', 256),
>              ('clip_range', 'lin_0.1'),
>              ('ent_coef', 0.01),
>              ('env_wrapper',
>               ['stable_baselines3.common.atari_wrappers.AtariWrapper']),
>              ('frame_stack', 4),
>              ('learning_rate', 'lin_2.5e-4'),
>              ('n_envs', 8),
>              ('n_epochs', 4),
>              ('n_steps', 128),
>              ('n_timesteps', 10000000.0),
>              ('policy', 'CnnPolicy'),
>              ('vf_coef', 0.5),
>              ('normalize', False)])

Link - https://huggingface.co/sb3/ppo-BreakoutNoFrameskip-v4

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is minimal and working
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
Hello,
you should use "BreakoutNoFrameskip-v4" for the env id.
Anyway, instructions to use it with the rl zoo are in the model card readme.

**Comment by indramal** [comment]
Thank you very much for your information. 

I think you need to edit the below article.

https://stable-baselines3.readthedocs.io/en/master/guide/integrations.html#hugging-face

**Comment by araffin** [comment]
what should be edited exactly?

**Comment by indramal** [comment]
Need to edit the following things in the below code.

-  `filename="ppo-CartPole-v1",` should be  `filename="ppo-CartPole-v1.zip",` (.zip should add otherwise file error showing)
- `eval_env = gym.make("CartPole-v1")` line should be `eval_env = gym.make("ppo-CartPole-v1") ` . I guess it.
- When using atari games need to add `eval_env = make_atari_env("Breakout-v4", n_envs=4, seed=0)`  `eval_env = VecFrameStack(eval_env , n_stack=4)` with  `from stable_baselines3.common.env_util import make_atari_env` `from stable_baselines3.common.vec_env import VecFrameStack` , otherwise it give shape error. (Like above code)

These are getting when I test. 

These are pages that need to edit,

- https://stable-baselines3.readthedocs.io/en/master/guide/integrations.html#hugging-face
- https://huggingface.co/sb3/demo-hf-CartPole-v1

When using atari games, `n_envs=` and `n_stack=` always need 4. Can not use 1. It give this shape error. 

> Error: Unexpected observation shape (1, 84, 84, 1) for Box environment, please use (4, 84, 84) or (n_env, 4, 84, 84) for the observation shape.

```
import gym

from huggingface_sb3 import load_from_hub
from stable_baselines3 import PPO
from stable_baselines3.common.evaluation import evaluate_policy

# Retrieve the model from the hub
## repo_id = id of the model repository from the Hugging Face Hub (repo_id = {organization}/{repo_name})
## filename = name of the model zip file from the repository
checkpoint = load_from_hub(
    repo_id="sb3/demo-hf-CartPole-v1",
    **filename="ppo-CartPole-v1"**,
)
model = PPO.load(checkpoint)

# Evaluate the agent and watch it
**eval_env = gym.make("CartPole-v1")**
mean_reward, std_reward = evaluate_policy(
    model, eval_env, render=True, n_eval_episodes=5, deterministic=True, warn=False
)
print(f"mean_reward={mean_reward:.2f} +/- {std_reward}")
```

**Comment by araffin** [comment]
> filename="ppo-CartPole-v1", should be filename="ppo-CartPole-v1.zip", (.zip should add otherwise file error showing)

you're right (@simoninithomas did something change or did we forget it from the beginning?)

> eval_env = gym.make("CartPole-v1") line should be eval_env = gym.make("ppo-CartPole-v1") . I guess it.

actually not, this line is correct. `"CartPole-v1"` is the env id and should be independent from the agent.

> When using atari games need to add eval_env = make_atari_env("Breakout-v4", n_envs=4, seed=0) eval_env = VecFrameStack(eval_env , n_stack=4) with from stable_baselines3.common.env_util import make_atari_env from stable_baselines3.common.vec_env import VecFrameStack , otherwise it give shape error. (Like above code)

well, it depends how the env was trained.
All the info about the hyperparameters are on each model page, and if you follow the instructions (i.e., use the RL Zoo), it will automatically recreate the env with the correct wrappers.

**Comment by simoninithomas** [comment]
Hey there 👋 . My bad, we changed it some months ago, we need to specify the zip file since we can 
load_from_hub any file. But indeed I forgot to update it on the SB3 documentation (it was updated on the Hub documentation and tutorials): https://huggingface.co/docs/hub/stable-baselines3#using-existing-models

I'm making a doc update PR that, thanks @indramal for pointing this out 🤗

**Comment by indramal** [comment]
@simoninithomas no problem
